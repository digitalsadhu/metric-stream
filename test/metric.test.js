'use strict';

const Metric = require('../lib/metric');
const util = require('util');

test('optional values return null when not provided', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
        description: 'Valid description',
        timestamp: 12345,
    });
    expect(metric.value).toBeNull();
    expect(metric.time).toBeNull();
    expect(metric.meta).toEqual({});
});

test('stringifying includes all keys', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
        description: 'Valid description',
        timestamp: 12345,
    });
    expect(metric).toMatchSnapshot();
});

test('omitting required keys results in undefined instead of null', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
    });
    expect(metric).toMatchSnapshot();
});

test('util.inspect includes all keys', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
        description: 'Valid description',
        timestamp: 12345,
        time: 12345,
        value: 123,
        meta: { key: 'value' },
    });
    expect(util.inspect(metric)).toMatchSnapshot();
});

test('toStringTag includes Metric in name', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
        description: 'Valid description',
        timestamp: 12345,
    });
    expect(`${metric}`).toBe(`Metric ${JSON.stringify(metric)}`);
});

test('toString outputs a full stringified object', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
        description: 'Valid description',
        timestamp: 12345,
    });
    expect(metric.toString()).toBe('[object Metric]');
});

test('parseInt on metric instance outputs null', () => {
    expect.hasAssertions();
    const metric = new Metric({
        name: 'valid_name',
        description: 'Valid description',
        timestamp: 12345,
    });
    expect(() => metric + 1).toThrowErrorMatchingSnapshot();
});
