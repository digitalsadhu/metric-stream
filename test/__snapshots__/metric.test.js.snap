// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omitting required keys results in undefined instead of null 1`] = `
Object {
  "description": undefined,
  "meta": Object {},
  "name": "valid_name",
  "time": null,
  "timestamp": undefined,
  "value": null,
}
`;

exports[`parseInt on metric instance outputs null 1`] = `"Invalid usage. Metric class instance cannot be treated as type 'default'"`;

exports[`stringifying includes all keys 1`] = `
Object {
  "description": "Valid description",
  "meta": Object {},
  "name": "valid_name",
  "time": null,
  "timestamp": 12345,
  "value": null,
}
`;

exports[`util.inspect includes all keys 1`] = `
"Metric { name: 'valid_name',
  description: 'Valid description',
  timestamp: 12345,
  value: 123,
  time: 12345,
  meta: { key: 'value' } }"
`;
